src/fs/driver/initfs/initfs_fops.c

```diff
@@ -0,0 +1,48 @@
+/**
+ * @
+ *
+ * @date Dec 23, 2019
+ * @author Anton Bondarev
+ */
+#include <stddef.h>
+#include <sys/types.h>
+#include <string.h>
+
+#include <fs/file_operation.h>
+#include <fs/file_desc.h>
+
+#include "initfs.h"
+
+static size_t initfs_read(struct file_desc *desc, void *buf, size_t size) {
+	struct initfs_file_info *fi;
+	off_t pos;
+
+	pos = file_get_pos(desc);
+	fi = file_get_inode_data(desc);
+
+	if (pos + size > file_get_size(desc)) {
+		size = file_get_size(desc) - pos;
+	}
+
+	memcpy(buf, (char *) (uintptr_t) (fi->start_pos + pos), size);
```

Comment:
Is this double cast `(char *) (uintptr_t)` really necessary?

Reply:
Yes, casting int to a pointer type isn't specified in standard